From 6c9443c0c1694600927dca29373d29fb741581f2 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Thu, 18 Jul 2024 02:08:07 +0530
Subject: [PATCH 3/3] Disable RTLD_DEEPBIND for Emscripten

RTLD_DEEPBIND gives symbol resolution preference to the loaded library
over the global scope, it's usually used to avoid symbol collision. However,
WebAssembly handles internal symbol resolution differently, and the flag is
is used on Linux systems and does not necessarily translate to other platforms
such as Emscripten.

This can be revisited after the upstream PR is resolved:
https://github.com/casadi/casadi/pull/3785

---
 casadi/core/casadi_os.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/casadi/core/casadi_os.cpp b/casadi/core/casadi_os.cpp
index d4718504..f285139c 100644
--- a/casadi/core/casadi_os.cpp
+++ b/casadi/core/casadi_os.cpp
@@ -115,7 +115,7 @@ handle_t open_shared_library(const std::string& lib, const std::vector<std::stri
             flag = RTLD_LAZY | RTLD_LOCAL;
         }
     #ifdef WITH_DEEPBIND
-    #ifndef __APPLE__
+    #if !defined(__APPLE__) && !defined(__EMSCRIPTEN__)
         flag |= RTLD_DEEPBIND;
     #endif
     #endif
-- 
2.39.3 (Apple Git-146)

